### PR TITLE
Consistent code with prototyped functions.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -79,70 +79,72 @@ Collector.prototype.computePointCountToSend = function(pointSizes, upperBound) {
   return Math.min(index, pointSizes.length); // But not if there were no points at all.
 };
 
-function notifyIfFlushed(collector, callback, err) {
-    if (!callback || collector._flushesInFlight > 0) {
+Collector.prototype._notifyIfFlushed = function(callback, err) {
+    var self = this;
```
Comment (IvanVergiliev) [inline index.js]: No need for the `self`.

File: index.js [hunk 2]
```diff
@@ -79,70 +79,72 @@ Collector.prototype.computePointCountToSend = function(pointSizes, upperBound) {
...
     if (!points || points.length === 0) {
         return;
     }
+    var self = this;
 
     // only send N points at a time to avoid making requests too large
     var spliceIndex;
-    if (collector._client.protocol == 'udp:') {
-      spliceIndex = collector.computePointCountToSend(collector._points.map(function (point) {
+    if (self._client.protocol == 'udp:') {
```
Comment (IvanVergiliev) [inline index.js:97]: A bunch of the `self`s here can be `this` instead - I don't think we have a policy on that so I guess either way works.

File: index.js [hunk 3]
```diff
@@ -79,70 +79,72 @@ Collector.prototype.computePointCountToSend = function(pointSizes, upperBound) {
...
+        self._flushSeries(key, series, callback);
     });
-    notifyIfFlushed(self, callback);
+    self._notifyIfFlushed(callback);
 };
 
-function getSeries(collector, name, reset) {
-    var series = collector._series[name];
+Collector.prototype._getSeries = function(name, reset) {
+    var self = this;
```
Comment (IvanVergiliev) [inline index.js]: No need for the `self`.

**Review by IvanVergiliev — COMMENTED** [review]
Thanks for fixing these!

File: index.js (hunk 2 above)
Comment (isoos) [reply index.js:97]: I've kept `self` when there was a `function()` callback, otherwise converted to `this`.

**Pull request merged**